src/mpyl/project.py

```diff
@@ -111,6 +111,60 @@ def from_config(values: dict):
         )
 
 
+@dataclass(frozen=True)
+class SecretKeyRef:
+    name: str
+    key: str
+    optional: bool
+    key_name: str = "secretKeyRef"
+
+    @staticmethod
+    def from_config(values: dict):
+        return SecretKeyRef(
+            name=values.get("name"),
+            key=values.get("key"),
+            optional=values.get("optional", False),
+        )
+
+
+@dataclass(frozen=True)
+class FieldRef:
+    field_path: str
+    key_name: str = "fieldRef"
+
+    @staticmethod
+    def from_config(values: dict):
+        return FieldRef(field_path=values.get("fieldPath"))
+
+
+@dataclass(frozen=True)
+class KeyValueRef:
+    key: str
+    value_from: Union[SecretKeyRef, FieldRef]
+
+    @staticmethod
+    def from_config(values: dict):
+        key = values.get("key")
+        value_from = values.get("valueFrom")
+
+        if not key or not value_from:
+            raise KeyError("KeyValueRef must have a key or valueFrom set")
+
+        key_name = list(value_from.keys())[0]
```

Comment:
are we actually sure there is always only 1 key?

Reply:
yes, according to the schema:
```yaml
        - name: SOME_SECRET_ENV
          valueFrom:
            secretKeyRef:
              key: password
              name: some-secret-name
              optional: false
        - name: KUBERNETES_NAMESPACE
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: metadata.namespace
```
so the list of keys are `list(secretKeyRef)` in the first case and `list(fieldRef)` in the second case